hearth/lib/hearth/event_stream/middleware/handlers.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module Hearth
+  module EventStream
+    module Middleware
+      # A middleware that configures the EventStream handlers
+      # on the request and response
+      # @api private
+      class Handlers
+        include Hearth::Middleware::Logging
+
+        # @param [Class] app The next middleware in the stack.
+        # @param [Boolean] request_events true if there are input events
+        # @param [Boolean] response_events true if there are output events
+        # @param [Class] async_output_class class to use for output.
+        #   Set only when the output is async.
+        # @param [Hearth::EventStream::HandlerBase] event_handler EventStream
+        #   handler for the operation to be used with request/response.
+        # @param [Module] message_encoding_module Module with protocol specific
+        #   message encoders/decoders.
+        #  arguments.
+        #
+        # rubocop:disable Metrics/ParameterLists
+        def initialize(
+          app,
+          request_events:, response_events:, async_output_class:,
+          event_handler:, message_encoding_module:
+        )
+          @app = app
+          @request_events = request_events
+          @response_events = response_events
+          @async_output_class = async_output_class
+          @event_handler = event_handler
+          @message_encoding_module = message_encoding_module
+        end
+        # rubocop:enable Metrics/ParameterLists
```

Comment:
We can probably just disable this cop or bump it since middleware has a lot of args by design.